kernel/include/onyx/rculist.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 Pedro Falcato
+ * This file is part of Onyx, and is released under the terms of the MIT License
+ * check LICENSE at the root directory for more information
+ *
+ * SPDX-License-Identifier: MIT
+ */
+#ifndef _ONYX_RCULIST_H
+#define _ONYX_RCULIST_H
+
+#include <onyx/list.h>
+#include <onyx/rcupdate.h>
+
+#define list_for_every_rcu(lh)                                                            \
+    for (struct list_head *l = __atomic_load_n(&(lh)->next, __ATOMIC_RELAXED); l != (lh); \
+         l = __atomic_load_n(&l->next, __ATOMIC_RELAXED))
```

Comment:
Nitpicking, but I'd add l to list of macro parameters so things are more explicit and there are no variables that appear out of thin air

Reply:
Yes, that makes sense, but i want list_for_every_rcu to be consistent with list_for_every for the time being. Ideally, I'd do a list_for_every LK/linux style where the container_of is done by the macro itself.
OTOH, this header isn't being used yet, I just committed it because it has been sitting on my working tree for way too long :)